framework/web/MultipartFormDataParser.php

```diff
@@ -0,0 +1,357 @@
+ * you are using this parser.
+ *
+ * Usage example:
+ *
+ * ```php
+ * use yii\web\UploadedFile;
+ *
+ * $restRequestData = Yii::$app->request->getBodyParams();
+ * $uploadedFile = UploadedFile::getInstancesByName('photo');
+ *
+ * $model = new Item();
+ * $model->populate($restRequestData);
+ * copy($uploadedFile->tempName, '/path/to/file/storage/photo.jpg');
+ * ```
+ *
+ * Note: although this parser fully emulates regular structure of the `$_FILES`, related temporary
+ * files, which are available via 'tmp_name' key, will not be recognized by PHP as uploaded ones.
+ * Thus functions like `is_uploaded_file()` and `move_uploaded_file()` will fail on them. This also
+ * means [[UploadedFile::saveAs()]] will fail as well.
+ *
+ * @property integer $uploadFileMaxSize upload file max size in bytes.
+ * @property integer $uploadFileMaxCount maximum upload files count.
+ *
+ * @author Paul Klimov <klimov.paul@gmail.com>
+ * @since 2.0.10
+ */
+class MultipartFormDataParser extends Object implements RequestParserInterface
+{
+    /**
+     * @var integer upload file max size in bytes.
+     */
+    private $_uploadFileMaxSize;
+    /**
+     * @var integer maximum upload files count.
+     */
+    private $_uploadFileMaxCount;
+
+
+    /**
+     * @return integer upload file max size in bytes.
+     */
+    public function getUploadFileMaxSize()
+    {
+        if ($this->_uploadFileMaxSize === null) {
+            $this->_uploadFileMaxSize = $this->getByteSize(ini_get('upload_max_filesize'));
+        }
+        return $this->_uploadFileMaxSize;
+    }
+
+    /**
+     * @param integer $uploadFileMaxSize upload file max size in bytes.
+     */
+    public function setUploadFileMaxSize($uploadFileMaxSize)
+    {
+        $this->_uploadFileMaxSize = $uploadFileMaxSize;
+    }
+
+    /**
+     * @return integer maximum upload files count.
+     */
+    public function getUploadFileMaxCount()
+    {
+        if ($this->_uploadFileMaxCount === null) {
+            $this->_uploadFileMaxCount = ini_get('max_file_uploads');
+        }
+        return $this->_uploadFileMaxCount;
+    }
+
+    /**
+     * @param integer $uploadFileMaxCount maximum upload files count.
+     */
+    public function setUploadFileMaxCount($uploadFileMaxCount)
+    {
+        $this->_uploadFileMaxCount = $uploadFileMaxCount;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function parse($rawBody, $contentType)
+    {
+        if (!empty($_POST) || !empty($_FILES)) {
+            // normal POST request is parsed by PHP automatically
+            return $_POST;
+        }
+
+        if (empty($rawBody)) {
+            return [];
+        }
+
+        if (!preg_match('/boundary=(.*)$/is', $contentType, $matches)) {
+            return [];
+        }
+        $boundary = $matches[1];
+
+        $bodyParts = preg_split('/-+' . preg_quote($boundary) . '/s', $rawBody);
+        array_pop($bodyParts); // last block always has no data
```

Comment:
You can pass `PREG_SPLIT_NO_EMPTY` as a flag to `preg_split` so it won't return empty elements.